File: MATTHEW/leetcode/AIcoding/2.Gemini-2.0/c_solutions/640-solve-the-equation.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

char* solveEquation(char* equation) {
    int x_coeff = 0, const_term = 0;
    int sign = 1;
    int num = 0;
    int i = 0;
    int len = strlen(equation);
    int left = 1;

    while (i < len) {
        if (equation[i] == '+') {
            const_term -= sign * num * left;
            num = 0;
            sign = 1;
            i++;
        } else if (equation[i] == '-') {
            const_term -= sign * num * left;
            num = 0;
            sign = -1;
            i++;
        } else if (equation[i] == '=') {
            const_term -= sign * num * left;
            num = 0;
            sign = 1;
            left = -1;
            i++;
        } else if (equation[i] == 'x') {
            if (num == 0) {
                x_coeff += sign * left;
            } else {
                x_coeff += sign * num * left;
            }
            num = 0;
            i++;
        } else {
            num = num * 10 + (equation[i] - '0');
            i++;
        }
    }
    const_term -= sign * num * left;

    if (x_coeff == 0) {
        if (const_term == 0) {
            return "Infinite solutions";
        } else {
            return "No solution";
        }
    } else {
        int ans = -const_term / x_coeff;
        char* result = (char*)malloc(20 * sizeof(char));
        sprintf(result, "x=%d", ans);
        return result;
    }
}

File: leetcode/AIcoding/2.Gemini-2.0/rust_solutions/640-solve-the-equation.rs
impl Solution {
    pub fn solve_equation(equation: String) -> String {
        let mut left_sign = 1;
        let mut coeff = 0;
        let mut const_term = 0;
        let mut num = 0;
        let mut sign = 1;

        for c in equation.chars() {
            match c {
                '+' => {
                    const_term += sign * num;
                    num = 0;
                    sign = 1;
                }
                '-' => {
                    const_term += sign * num;
                    num = 0;
                    sign = -1;
                }
                'x' => {
                    if num == 0 {
                        coeff += left_sign * sign * 1;
                    } else {
                        coeff += left_sign * sign * num;
                    }
                    num = 0;
                    sign = 1;
                }
                '=' => {
                    const_term += sign * num;
                    num = 0;
                    left_sign = -1;
                    sign = 1;
                }
                _ => {
                    num = num * 10 + (c as i32 - '0' as i32) as i32;
                }
            }
        }

        const_term += sign * num;

        if coeff == 0 {
            if const_term == 0 {
                return "Infinite solutions".to_string();
            } else {
                return "No solution".to_string();
            }
        } else {
            let ans = -const_term / coeff;
            return format!("x={}", ans);
        }
    }
}